frontend/src/employee-frontend/components/unit/tab-calendar/discussion-surveys/survey-editor/DiscussionSurveyForm.tsx

```diff
@@ -0,0 +1,304 @@
+              } else {
+                navigate(
+                  `/units/${unitId}/groups/${groupId}/discussion-reservation-surveys`,
+                  { replace: true }
+                )
+              }
+            },
+            label: t.discussionReservation.cancelConfirmation.continueButton
+          }}
+          text={t.discussionReservation.cancelConfirmation.text}
+        />
+      )}
+
+      {eventData ? (
+        <H2>{title.state}</H2>
+      ) : (
+        <H2>{t.discussionReservation.surveyCreate}</H2>
+      )}
+      <H3>{t.discussionReservation.surveyBasicsTitle}</H3>
+
+      <SurveyFormSectionGroup>
+        <SurveyFormFieldGroup>
+          <Label>{t.discussionReservation.surveySubject}</Label>
+          <WidthLimiter>
+            <TextAreaF
+              bind={title}
+              info={info('title')}
+              placeholder={t.discussionReservation.surveySubjectPlaceholder}
+              maxLength={30}
+              data-qa="survey-title-input"
+            />
+          </WidthLimiter>
+        </SurveyFormFieldGroup>
+
+        <SurveyFormFieldGroup>
+          <ExpandingInfo info={t.discussionReservation.surveySummaryInfo}>
+            <Label>{t.discussionReservation.surveySummary}</Label>
+          </ExpandingInfo>
+          <WidthLimiter>
+            <TextAreaF
+              bind={description}
+              info={info('description')}
+              placeholder={t.discussionReservation.surveySummaryPlaceholder}
+              data-qa="survey-description-input"
+            />
+          </WidthLimiter>
+        </SurveyFormFieldGroup>
+      </SurveyFormSectionGroup>
+      <SurveyFormSectionGroup>
+        <H3>{t.discussionReservation.surveyInviteeTitle}</H3>
+        <SurveyFormFieldGroup>
+          <WidthLimiter>
+            <Label>{t.discussionReservation.surveyInvitees}</Label>
+            <TreeDropdown
+              tree={attendees.state}
+              onChange={attendees.set}
+              data-qa="survey-attendees-select"
+              placeholder={
+                i18n.unit.calendar.events.create.attendeesPlaceholder
+              }
+            />
+          </WidthLimiter>
+        </SurveyFormFieldGroup>
+      </SurveyFormSectionGroup>
+
+      <FixedSpaceRow justifyContent="flex-start" alignItems="center">
+        <Button
+          text={t.discussionReservation.cancelButton}
+          onClick={() => setCancelConfirmModalVisible(true)}
+          data-qa="cancel-button"
+        />
+        {eventData ? (
+          <MutateButton
+            primary
+            mutation={updateCalendarEventMutation}
+            text={t.discussionReservation.saveSurveyButton}
+            disabled={!isBasicInfoValid}
+            onSuccess={() =>
+              navigate(
+                `/units/${unitId}/groups/${groupId}/discussion-reservation-surveys/${eventData.id}`
+              )
+            }
+            onClick={() => {
+              const values = form.value()
```

Comment:
If I remember correctly, calling `.value()` throws if the form is not valid, so you might want to check `.isValid()` before and possibly return cancelMutation, even if the disabled prop might mostly already handle that scenario